src/main/java/woohakdong/server/api/service/admin/auth/AdminAuthService.java

```diff
@@ -0,0 +1,121 @@
+import woohakdong.server.api.controller.admin.auth.dto.AdminJoinRequest;
+import woohakdong.server.api.controller.admin.auth.dto.AdminLoginRequest;
+import woohakdong.server.api.controller.auth.dto.LoginResponse;
+import woohakdong.server.common.exception.CustomException;
+import woohakdong.server.common.security.jwt.CustomUserDetails;
+import woohakdong.server.common.security.jwt.JWTUtil;
+import woohakdong.server.domain.member.Member;
+import woohakdong.server.domain.member.MemberRepository;
+import woohakdong.server.domain.refreshToken.RefreshToken;
+import woohakdong.server.domain.refreshToken.RefreshTokenRepository;
+
+import java.util.Date;
+
+import static woohakdong.server.common.exception.CustomErrorInfo.*;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class AdminAuthService {
+
+    private final MemberRepository memberRepository;
+    private final RefreshTokenRepository refreshTokenRepository;
+    private final JWTUtil jwtUtil;
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    @Transactional
+    public LoginResponse login(AdminLoginRequest loginRequest) {
+        Member admin = memberRepository.findByMemberProvideId(loginRequest.username())
+                .orElseThrow(() -> new CustomException(ADMIN_MEMBER_ID_NOT_FOUND));
+
+        if (!passwordEncoder.matches(loginRequest.password(), admin.getMemberPassword())) {
+            throw new CustomException(INVALID_ADMIN_PASSWORD);
+        }
+
+        String access = jwtUtil.createJwt("access", admin.getMemberProvideId(), admin.getMemberRole(), 600000L);
+        String refresh = jwtUtil.createJwt("refresh", admin.getMemberProvideId(), admin.getMemberRole(), 86400000L);
+
+        addRefreshEntity(admin.getMemberProvideId(), refresh, 86400000L);
+
+        return LoginResponse.builder()
+                .accessToken(access)
+                .refreshToken(refresh)
+                .build();
+    }
+
+    @Transactional
+    public void createAdmin(AdminJoinRequest joinRequest) {
+        if (memberRepository.findByMemberProvideId(joinRequest.username()).isPresent()) {
+            throw new CustomException(ADMIN_USERNAME_IS_ALREADY_USED);
+        }
+
+        String encryptedPassword = passwordEncoder.encode("1234");
+
+        Member admin = Member.builder()
+                .memberProvideId(joinRequest.username())
+                .memberName(joinRequest.name())
+                .memberRole("ADMIN_ROLE")
+                .memberEmail(joinRequest.email())
+                .memberPassword(encryptedPassword)
+                .build();
+
+        memberRepository.save(admin);
+    }
+
+    @Transactional
+    public void updateAdmin(AdminInfoUpdateRequest updateRequest) {
+        Member admin = getMemberFromJwtInformation();
+
+        if (memberRepository.findByMemberProvideId(updateRequest.username()).isPresent()) {
+            throw new CustomException(ADMIN_USERNAME_IS_ALREADY_USED);
+        }
```

Comment:
꼼꼼하게 체크하신 것 좋습니다!
서비스에서 반복적으로 예외 던지는 것보다는 Repository 계층으로 해당 로직 옮기는게 좋을 것 같아요

```suggestion
        memberRepository.checkByMemberProvideId(updateRequest.username());
```

Reply:
이 부분은 반복적으로 예외를 던진다기 보다 findByMemberProvideId 이 로직을 수행할 때 여러 예외가 터질 수 있기 때문에 일부러 서비스 로직에 위치 시켰습니다.